doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+                    "status": "false"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "100"
+                }
+            ]
+        },
+        {
+            "name": "all FAN and PSU presence",
+            "conditions": [
+                {
+                    "type": "fan.all.presence"
+                },
+                {
+                    "type": "psu.all.presence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "true"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "60"
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### Invalid format policy file
+
+In invalid_format_policy.json, the file content is not valid JSON at all. A file with content "invalid" should be good for test purpose.
+
+#### Invalid value policy file
+
+In invalid_value_policy.json, the file content contains minus value of target speed. We couldn't cover all invalid value here because there are too many possibilities. The major purpose of this configuration file is to verify thermal control daemon would not crash while loading such an invalid configuration files. For other negative test cases, they are already covered by unit test. The invalid_value_policy.json content is like:
+
+```json
+{
+    "info_types": [
+        {
+            "type": "fan_info"
+        },
+        {
+            "type": "psu_info"
+        }
+    ],
+    "policies": [
+        {
+            "name": "any PSU absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "-1"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Test Cases
+
+### Show FAN Status Test
+
+Show FAN status test verifies that all FAN related information can be shown correctly via `show platform fanstatus`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink FAN related sysfs and fill fake data for "presence", "speed", "status", "target_speed", "led status".
+3. Issue command `show platform fanstatus`.
+4. Record the command output.
+5. Verify that command output matches the fake data.
+6. Link FAN related sysfs.
+
+### Show Thermal Status Test
+
+Show thermal status test verifies that all thermal related information can be shown correctly via `show platform temperature`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink thermal related sysfs and fill fake data for "temperature", "high_threshold", "high_critical_threshold".
+3. Issue command `show platform temperature`.
+4. Record the command output.
+5. Verify that command output matches the fake data.
+6. Link thermal related sysfs.
```

Comment:
Re-link by "ln -s xxx" works, I tested it today. No need use "loganalyzer" because the "Show  Thermal Status Test" only mocks correct data, will add "loganalyzer" to negative test case.